### ISSUE TITLE
Blank screen with sizing_mode set to 'stretch_both' in Safari

### ISSUE THREAD

**Issue by wcatron** [bokeh/bokeh#5782]
With sizing mode set to `stretch_both` when the page first loads in Safari it's blank. Once there's a resize event it works. I tested the same code with scale_both and it works fine.

**Comment by iaroslav-ai** [comment]
I have the same issue on Chrome and Firefox in Ubuntu. The browser console output is as follows:

```
[bokeh] setting log level to: 'info'
bokeh.min.js?v=9d3af13…:1 [bokeh] Websocket connection 0 is now open
bokeh.min.js?v=9d3af13…:9 Error rendering Bokeh items  TypeError: Cannot read property 'clientHeight' of null
    at e.r.PlotView.e.get_width_height (bokeh.min.js?v=9d3af13…:18)
    at e.r.PlotView.e.render (bokeh.min.js?v=9d3af13…:18)
    at p (bokeh.min.js?v=9d3af13…:7)
    at c (bokeh.min.js?v=9d3af13…:7)
    at i (bokeh.min.js?v=9d3af13…:7)
    at e.r.Events.trigger (bokeh.min.js?v=9d3af13…:7)
    at e.setv (bokeh.min.js?v=9d3af13…:4)
    at e.r.HasProps.e.setv (bokeh.min.js?v=9d3af13…:7)
    at e.<anonymous> (bokeh.min.js?v=9d3af13…:18)
    at p (bokeh.min.js?v=9d3af13…:7)
bokeh.min.js?v=9d3af13…:18 Uncaught TypeError: Cannot read property 'clientHeight' of null
    at e.r.PlotView.e.get_width_height (bokeh.min.js?v=9d3af13…:18)
    at e.r.PlotView.e.render (bokeh.min.js?v=9d3af13…:18)
    at e.<anonymous> (bokeh.min.js?v=9d3af13…:16)
    at p (bokeh.min.js?v=9d3af13…:7)
    at c (bokeh.min.js?v=9d3af13…:7)
    at i (bokeh.min.js?v=9d3af13…:7)
    at t.r.Events.trigger (bokeh.min.js?v=9d3af13…:7)
    at t.r.Document.t._resize (bokeh.min.js?v=9d3af13…:9)
    at t.r.Document.t.resize (bokeh.min.js?v=9d3af13…:9)
    at bokeh.min.js?v=9d3af13…:140
r.PlotView.e.get_width_height @ bokeh.min.js?v=9d3af13…:18
r.PlotView.e.render @ bokeh.min.js?v=9d3af13…:18
(anonymous) @ bokeh.min.js?v=9d3af13…:16
p @ bokeh.min.js?v=9d3af13…:7
c @ bokeh.min.js?v=9d3af13…:7
i @ bokeh.min.js?v=9d3af13…:7
r.Events.trigger @ bokeh.min.js?v=9d3af13…:7
r.Document.t._resize @ bokeh.min.js?v=9d3af13…:9
r.Document.t.resize @ bokeh.min.js?v=9d3af13…:9
(anonymous) @ bokeh.min.js?v=9d3af13…:140
```

Apparently "clientHeight" property is missing somewhere.

**Comment by mattpap** [comment]
@iaroslav-ai, I wasn't able to reproduce this. What version of bokehjs do you use? Also, can you provide code that gives you this error?

**Comment by bryevdv** [comment]
@mattpap FYI I am personally fine with closing any issue that does not provide the information requested in the issue template with `invalid`

**Comment by wcatron** [comment]
bokeh: 0.12.4
Python: 2.7.10
Notebook: Macbook Pro
OS: Sierra 10.12.3
Browser: Safari 10.0.3
I also had a sub for lunch.

I already listed expected vs. actual behavior.

`
p = figure(title='', x_axis_label='x', y_axis_label='y',sizing_mode='stretch_both', width=7, height=3, x_axis_type="datetime", tools="pan,undo,redo,reset,wheel_zoom,box_zoom,save", toolbar_location="above")
# Add anything to this.
`

Resulting HTML: https://jsbin.com/qoficayiba/edit?html,js,output

FYI. imho Closing an issue and marking it as invalid just because someone did not fill out your anal list of requests will not encourage people to contribute to your project and report issues.

**Comment by bryevdv** [comment]
> FYI. imho Closing an issue and marking it as invalid just because someone did not fill out your anal list of requests will not encourage people to contribute to your project and report issues.

It's impossible to work on issues that don't have necessary information, so leaving them open is literally pointless. If you want free help and support on OSS projects but can't be bothered to provide the information needed for OSS dev to help you then I don't know what to tell you.

**Comment by iaroslav-ai** [comment]
@mattpap hi sorry I commented a bit in a rush, it appears that I have an issue with something else. If I will figure that it has something to do with bokeh itself, I will open a new issue, thx.

**Comment by wcatron** [comment]
I did not open this issue looking for "free help and support" I opened it because there's an issue with the library. I understand needing information for support, I've provided support and I know "Hey, ____ isn't working." is not enough information and gets very frustrating. Again, imho there's a balance; I think closing and marking issues as invalid before giving developers a chance to fully evaluate them is too far. You disagree.

----

In terms of this issue: I changed the css to this
```css
.bk-root {
    position: absolute;
}
```
JSBin: https://jsbin.com/salukaqiha/1/edit?html,css,js,output

And that worked. I think there's an issue with using relative with height 100%. I had to simplify the margins and everything to get it to look the same. Perhaps that whole nested structure can be simplified.
